pkg/registries/ecr/ecr.go

```diff
@@ -180,51 +117,37 @@ func newRegistry(integration *storage.ImageIntegration, disableRepoList bool) (*
 	reg := &ecr{
 		config:      conf,
 		integration: integration,
-		// docker endpoint
-		endpoint:        fmt.Sprintf("%s.dkr.ecr.%s.amazonaws.com", conf.GetRegistryId(), conf.GetRegion()),
-		disableRepoList: disableRepoList,
 	}
+	endpoint := fmt.Sprintf("%s.dkr.ecr.%s.amazonaws.com", conf.GetRegistryId(), conf.GetRegion())
 	// If the ECR configuration provides Authorization Data, we do not initialize an
 	// ECR client, but instead, we create the registry immediately since the
 	// Authorization Data payload provides the credentials statically.
+	var cfg *docker.Config
 	if authData := conf.GetAuthorizationData(); authData != nil {
-		expiresAt, err := protobuftypes.TimestampFromProto(authData.GetExpiresAt())
-		if err != nil {
-			return nil, errors.New("invalid authorization data")
-		}
-		if err = reg.setRegistry(authData.GetUsername(), authData.GetPassword(), expiresAt); err != nil {
-			return nil, errors.Wrap(err, "failed to create registry client")
+		cfg = &docker.Config{
+			Username:        authData.GetUsername(),
+			Password:        authData.GetPassword(),
+			Endpoint:        endpoint,
+			DisableRepoList: disableRepoList,
 		}
 	} else {
-		service, err := createECRClient(conf)
+		client, err := createECRClient(conf)
 		if err != nil {
+			log.Error("Failed to create client: ", err)
```

Comment:
nit: maybe be more specific about which client, `ECR` (even though the log line will say this is in the ecr.go file)